politeiawww/backend.go

```diff
@@ -106,6 +106,24 @@ func decodeBackendProposalMetadata(payload []byte) (*BackendProposalMetadata, er
 	return &md, nil
 }
 
+// Compare supplied public key against the one stored in the user database
+// It will return the curent active public key if there are no errors
+func checkPublicKey(user *database.User, pk string) ([identity.PublicKeySize]byte, error) {
+	id, ok := database.ActiveIdentity(user.Identities)
+	if !ok {
+		return [identity.PublicKeySize]byte{}, www.UserError{
+			ErrorCode: www.ErrorStatusNoPublicKey,
+		}
+	}
+
+	if hex.EncodeToString(id[:]) != pk {
+		return [identity.PublicKeySize]byte{}, www.UserError{
+			ErrorCode: www.ErrorStatusInvalidSigningKey,
+		}
+	}
+	return id, nil
+}
+
 // Check an incomming signature against the specified user's pubkey.
 func checkSig(user *database.User, signature string, elements ...string) error {
```

Comment:
This function no longer has to fetch the public key from the user since it's being checked against the user's active identities in `checkPublicKey`. Instead of taking the user as input, it should take the public key.

Reply:
Yes, it makes sense. I will make the change as soon as possible.